spec/models/move_spec.rb

```diff
@@ -543,6 +543,63 @@
     end
   end
 
+  describe '#determine_supplier' do
+    subject(:determined_supplier) { move.determine_supplier }
+
+    let(:location) { create(:location, :police) }
+    let(:old_supplier) { create(:supplier) }
+    let(:new_supplier) { build(:supplier) }
+    let(:date) { Date.today }
+    let(:date_from) { Date.today }
+    let(:move) { create(:move, :cancelled, from_location: location, to_location: nil, supplier: old_supplier, date: date, date_from: date_from) }
+
+    before do
+      supplier_chooser = instance_double('SupplierChooser', call: new_supplier)
+      allow(SupplierChooser).to receive(:new).and_return(supplier_chooser)
+    end
+
+    context 'when date and date_from have not changed' do
+      it 'returns existing supplier' do
+        expect(determined_supplier).to eq old_supplier
+      end
+
+      it 'does not call SupplierChooser service' do
+        determined_supplier
+        expect(SupplierChooser).not_to have_received(:new)
+      end
+    end
+
+    context 'when date is present and has changed' do
+      let(:date_from) { nil }
```

Comment:
we need to set date_from to nil?
this makes the test a bit confusing: "when date is present and has changed"

Reply:
Yes because the implementation will use the `date` in preference if it's present.